From 59ba0c8e60b74ad7023a4b01487d7b0fa60c074a Mon Sep 17 00:00:00 2001
From: Vladimir Ermakov <vooon341@gmail.com>
Date: Wed, 23 Jul 2025 13:14:05 +0200
Subject: [PATCH 1/3] cargo: fix for client only build

- delete debian specific cargo config
- disable build of the backup server, leave only client
- set dependencies git urls

Signed-off-by: Vladimir Ermakov <vooon341@gmail.com>
---
 .cargo/config.toml |   8 ----
 Cargo.toml         | 100 +++++++++++++++++++++++----------------------
 2 files changed, 51 insertions(+), 57 deletions(-)
 delete mode 100644 .cargo/config.toml

diff --git a/.cargo/config.toml b/.cargo/config.toml
deleted file mode 100644
index a439c97b..00000000
--- a/.cargo/config.toml
+++ /dev/null
@@ -1,8 +0,0 @@
-[source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
-[source.crates-io]
-replace-with = "debian-packages"
-
-[profile.release]
-debug=true
diff --git a/Cargo.toml b/Cargo.toml
index ea413328..e532ccae 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,10 +39,10 @@ members = [
     "pbs-tape",
     "pbs-tools",
 
-    "proxmox-backup-banner",
+    #"proxmox-backup-banner",
     "proxmox-backup-client",
     "proxmox-file-restore",
-    "proxmox-restore-daemon",
+    #"proxmox-restore-daemon",
 
     "pxar-bin",
 ]
@@ -53,8 +53,8 @@ path = "src/lib.rs"
 
 [workspace.dependencies]
 # proxmox workspace
-proxmox-apt = { version = "0.99", features = [ "cache" ] }
-proxmox-apt-api-types = "2"
+#proxmox-apt = { version = "0.99", features = [ "cache" ] }
+#proxmox-apt-api-types = "2"
 proxmox-async = "0.5"
 proxmox-auth-api = "1"
 proxmox-base64 = "1"
@@ -68,10 +68,10 @@ proxmox-human-byte = "1"
 proxmox-io = "1.0.1" # tools and client use "tokio" feature
 proxmox-lang = "1.1"
 proxmox-log = "1"
-proxmox-ldap = "1"
-proxmox-metrics = "1"
+#proxmox-ldap = "1"
+#proxmox-metrics = "1"
 proxmox-notify = "1"
-proxmox-openid = "1"
+#proxmox-openid = "1"
 proxmox-rest-server = { version = "1", features = [ "templates" ] }
 # some use "cli", some use "cli" and "server", pbs-config uses nothing
 proxmox-router = { version = "3.2.2", default-features = false }
@@ -85,7 +85,7 @@ proxmox-serde = "1"
 proxmox-shared-cache = "1"
 proxmox-shared-memory = "1"
 proxmox-sortable-macro = "1"
-proxmox-subscription = { version = "1", features = [ "api-types" ] }
+#proxmox-subscription = { version = "1", features = [ "api-types" ] }
 proxmox-sys = "1"
 proxmox-systemd = "1"
 proxmox-tfa = { version = "6", features = [ "api", "api-types" ] }
@@ -96,7 +96,7 @@ pbs-api-types = "1.0.2"
 
 # other proxmox crates
 pathpatterns = "1"
-proxmox-acme = "1"
+#proxmox-acme = "1"
 pxar = "1"
 
 # PBS workspace
@@ -113,7 +113,7 @@ pbs-tools = { path = "pbs-tools" }
 # regular crates
 anyhow = "1.0"
 async-trait = "0.1.56"
-apt-pkg-native = "0.3.2"
+#apt-pkg-native = "0.3.2"
 bitflags = "2.4"
 bytes = "1.0"
 cidr = "0.3"
@@ -206,8 +206,8 @@ zstd.workspace = true
 #valgrind_request = { git = "https://github.com/edef1c/libvalgrind_request", version = "1.1.0", optional = true }
 
 # proxmox workspace
-proxmox-apt.workspace = true
-proxmox-apt-api-types.workspace = true
+#proxmox-apt.workspace = true
+#proxmox-apt-api-types.workspace = true
 proxmox-async.workspace = true
 proxmox-auth-api = { workspace = true, features = [ "api", "pam-authenticator" ] }
 proxmox-base64.workspace = true
@@ -219,11 +219,11 @@ proxmox-human-byte.workspace = true
 proxmox-io.workspace = true
 proxmox-lang.workspace = true
 proxmox-log.workspace = true
-proxmox-ldap.workspace = true
-proxmox-metrics.workspace = true
+#proxmox-ldap.workspace = true
+#proxmox-metrics.workspace = true
 proxmox-notify = { workspace = true, features = [ "pbs-context" ] }
-proxmox-openid.workspace = true
-proxmox-rest-server = { workspace = true, features = [ "rate-limited-stream" ] }
+#proxmox-openid.workspace = true
+#proxmox-rest-server = { workspace = true, features = [ "rate-limited-stream" ] }
 proxmox-router = { workspace = true, features = [ "cli", "server"] }
 proxmox-s3-client.workspace = true
 proxmox-schema = { workspace = true, features = [ "api-macro" ] }
@@ -232,7 +232,7 @@ proxmox-serde = { workspace = true, features = [ "serde_json" ] }
 proxmox-shared-cache.workspace = true
 proxmox-shared-memory.workspace = true
 proxmox-sortable-macro.workspace = true
-proxmox-subscription.workspace = true
+#proxmox-subscription.workspace = true
 proxmox-sys = { workspace = true, features = [ "timer" ] }
 proxmox-systemd.workspace = true
 proxmox-tfa.workspace = true
@@ -242,7 +242,7 @@ proxmox-worker-task.workspace = true
 pbs-api-types.workspace = true
 
 # in their respective repo
-proxmox-acme.workspace = true
+#proxmox-acme.workspace = true
 pxar.workspace = true
 
 # proxmox-backup workspace/internal crates
@@ -259,46 +259,48 @@ proxmox-rrd-api-types.workspace = true
 # Local path overrides
 # NOTE: You must run `cargo update` after changing this for it to take effect!
 [patch.crates-io]
-#pbs-api-types = { path = "../proxmox/pbs-api-types" }
+pbs-api-types = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
 #proxmox-acme = { path = "../proxmox/proxmox-acme" }
 #proxmox-apt = { path = "../proxmox/proxmox-apt" }
 #proxmox-apt-api-types = { path = "../proxmox/proxmox-apt-api-types" }
-#proxmox-async = { path = "../proxmox/proxmox-async" }
-#proxmox-auth-api = { path = "../proxmox/proxmox-auth-api" }
-#proxmox-base64 = { path = "../proxmox/proxmox-base64" }
-#proxmox-borrow = { path = "../proxmox/proxmox-borrow" }
-#proxmox-compression = { path = "../proxmox/proxmox-compression" }
-#proxmox-config-digest = { path = "../proxmox/proxmox-config-digest" }
-#proxmox-daemon = { path = "../proxmox/proxmox-daemon" }
-#proxmox-fuse = { path = "../proxmox-fuse" }
-#proxmox-http = { path = "../proxmox/proxmox-http" }
-#proxmox-human-byte = { path = "../proxmox/proxmox-human-byte" }
-#proxmox-io = { path = "../proxmox/proxmox-io" }
-#proxmox-lang = { path = "../proxmox/proxmox-lang" }
-#proxmox-log = { path = "../proxmox/proxmox-log" }
+proxmox-async = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-auth-api = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-base64 = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-borrow = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-compression = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-config-digest = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-daemon = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-fuse = { git = "https://git.proxmox.com/git/proxmox-fuse.git", branch="master" }
+proxmox-http = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-human-byte = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-io = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-lang = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-log = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
 #proxmox-ldap = { path = "../proxmox/proxmox-ldap" }
 #proxmox-metrics = { path = "../proxmox/proxmox-metrics" }
-#proxmox-notify = { path = "../proxmox/proxmox-notify" }
+proxmox-notify = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
 #proxmox-openid = { path = "../proxmox/proxmox-openid" }
 #proxmox-rest-server = { path = "../proxmox/proxmox-rest-server" }
-#proxmox-router = { path = "../proxmox/proxmox-router" }
-#proxmox-rrd = { path = "../proxmox/proxmox-rrd" }
-#proxmox-rrd-api-types = { path = "../proxmox/proxmox-rrd-api-types" }
-#proxmox-schema = { path = "../proxmox/proxmox-schema" }
-#proxmox-section-config = { path = "../proxmox/proxmox-section-config" }
-#proxmox-serde = { path = "../proxmox/proxmox-serde" }
-#proxmox-shared-memory = { path = "../proxmox/proxmox-shared-memory" }
-#proxmox-sortable-macro = { path = "../proxmox/proxmox-sortable-macro" }
+proxmox-router = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-rrd = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-rrd-api-types = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-s3-client = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-schema = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-section-config = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-serde = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-shared-cache = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-shared-memory = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-sortable-macro = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
 #proxmox-subscription = { path = "../proxmox/proxmox-subscription" }
-#proxmox-sys = { path = "../proxmox/proxmox-sys" }
-#proxmox-systemd = { path = "../proxmox/proxmox-systemd" }
-#proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
-#proxmox-time = { path = "../proxmox/proxmox-time" }
-#proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
-#proxmox-worker-task = { path = "../proxmox/proxmox-worker-task" }
+proxmox-sys = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-systemd = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-tfa = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-time = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-uuid = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
+proxmox-worker-task = { git = "https://git.proxmox.com/git/proxmox.git", branch="master" }
 
-#pathpatterns = {path = "../pathpatterns" }
-#pxar = { path = "../pxar" }
+pathpatterns = { git = "https://git.proxmox.com/git/pathpatterns.git", branch="master" }
+pxar = { git = "https://git.proxmox.com/git/pxar.git", branch="master" }
 
 [features]
 default = []
-- 
2.50.1

